core/src/io.rs

```diff
@@ -157,32 +149,40 @@ impl<T: Metadata> MetaIoHandler<T> {
 
 /// Simplified `IoHandler` with no `Metadata` associated with each request.
 #[derive(Default)]
-pub struct IoHandler(MetaIoHandler<()>);
+pub struct IoHandler<M: Metadata = ()>(MetaIoHandler<M>);
```

Comment:
To support writing test cases not using metadata easier:

```rust
assert_eq!(io.handle_request_sync(&request), Some(response));
```

and for meta you need to do:
```rust
assert_eq!((*io).handle_request_sync(&request, meta), Some(response));
```
or just use `MetaIoHandler` directly.